x/oracle/types/msgs.go

```diff
@@ -168,3 +172,115 @@ func (msg MsgDelegateFeedConsent) ValidateBasic() error {
 
 	return nil
 }
+
+// NewMsgUpdateParams will creates a new MsgUpdateParams instance
+func NewMsgUpdateParams(authority, title, description string, keys []string, changes Params) *MsgGovUpdateParams {
+	return &MsgGovUpdateParams{
+		Title:       title,
+		Description: description,
+		Authority:   authority,
+		Keys:        keys,
+		Changes:     changes,
+	}
+}
+
+// Type implements Msg interface
+func (msg MsgGovUpdateParams) Type() string { return sdk.MsgTypeURL(&msg) }
+
+// String implements the Stringer interface.
+func (msg MsgGovUpdateParams) String() string {
+	out, _ := yaml.Marshal(msg)
+	return string(out)
+}
+
+// GetSignBytes implements Msg
+func (msg MsgGovUpdateParams) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(&msg)
+	return sdk.MustSortJSON(bz)
+}
+
+// GetSigners implements Msg
+func (msg MsgGovUpdateParams) GetSigners() []sdk.AccAddress {
+	return checkers.Signers(msg.Authority)
+}
+
+// ValidateBasic implements Msg
+func (msg MsgGovUpdateParams) ValidateBasic() error {
+	if err := checkers.ValidateProposal(msg.Title, msg.Description, msg.Authority); err != nil {
+		return err
+	}
+
+	for _, key := range msg.Keys {
+		switch key {
+		case string(KeyVotePeriod):
+			if msg.Changes.VotePeriod == 0 {
+				return fmt.Errorf("oracle parameter VotePeriod must be > 0")
+			}
+
+		case string(KeyVoteThreshold):
+			if msg.Changes.VoteThreshold.LTE(sdk.NewDecWithPrec(33, 2)) {
+				return fmt.Errorf("oracle parameter VoteThreshold must be greater than 33 percent")
+			}
+
+		case string(KeyRewardBands):
+			for _, v := range msg.Changes.RewardBands {
+				if v.RewardBand.GT(sdk.OneDec()) || v.RewardBand.IsNegative() {
+					return fmt.Errorf("oracle parameter RewardBand must be between [0, 1]")
+				}
+				if len(v.SymbolDenom) == 0 {
+					return fmt.Errorf("oracle parameter RewardBand must have SymbolDenom")
+				}
+			}
+
+		case string(KeyRewardDistributionWindow):
+			if msg.Changes.RewardDistributionWindow == 0 {
+				return fmt.Errorf("oracle parameter RewardDistributionWindow must be > 0")
+			}
+
+		case string(KeyAcceptList):
+			for _, denom := range msg.Changes.AcceptList {
+				if len(denom.BaseDenom) == 0 {
```

Comment:
Could we just have a generic validation function for `denom` objects? that way we have less duplicated code

Reply:
[8b77db2](https://github.com/ojo-network/ojo/pull/101/commits/8b77db262fc2043888cb09428ae62edd5561bfcd)